### DISCUSSION TITLE
Cannot boot up when secure-boot enabled

### DISCUSSION THREAD

**Discussion by Tzu-Chin** [ipxe/ipxe#573] [Q&A] [answered]
Hello guys,

I manage to set up an environment for diskless boot and would like to go with the secure-boot feature. I used the command to build the EFI file and sign it.
$ make bin-x86_64-efi EMBED=my_script

My script is quite simple.
<img width="256" alt="2" src="https://user-images.githubusercontent.com/53668363/150751043-56161ac8-4ac0-40b5-80b3-55a7038bba11.PNG">

For narrowing down the scope, I disabled the secure-boot first. Everything goes as expected.
![3](https://user-images.githubusercontent.com/53668363/150751280-6a70489d-4d3e-4f31-93ed-155c6d634737.jpg)

However, when I enable the secure-boot item, I got an error message "Exec format error". 
![4](https://user-images.githubusercontent.com/53668363/150751370-3fb3e69f-e8cd-4482-affa-1ba5b004313e.jpg)

I was surprised about the outcome, because the files in the FTP are exactly the same. How come the format become incorrect as enabled the secure boot?
Any suggestions are really appreciated.

Thank you.

**Comment by NiKiZe** [comment, marked as answer]
You might want to post the message as text, including the ipxe.org URL.

You can see that iPXE boot just fine.
But is vmlinuz signed with a valid key?

**Reply by Tzu-Chin** [reply]
Thanks a lot. I got a misunderstanding about that. 
Yes, after signing the EFI and file and vmlinuz, it can work as expected.
